src/charm.py

```diff
@@ -399,6 +404,19 @@ def _on_update_status(self, _) -> None:
         else:
             self.unit.status = WaitingStatus()
 
+    def _on_get_client_connection_data(self, event: ActionEvent) -> None:
```

Comment:
misnomer: it is partial connection data. should we also include host and port in the results?

Reply:
For completion, yes. I didn't bother since usage on tests rely on helpers for getting unit ip/addres, which might be not resolvable from outside the cluster